cmd/nitro/nitro.go

```diff
@@ -882,6 +916,36 @@ func (c *NodeConfig) Validate() error {
 	if err := c.Execution.Validate(); err != nil {
 		return err
 	}
+	if !c.ExecutionNode {
+		if c.Node.ExecutionRPCClient.URL == "" {
+			return errors.New("starting consensus only node with empty executionRPCClient url")
+		}
+		if c.ConsensusExecutionUseRPC {
+			return errors.New("starting without an execution node but have ConsensusExecutionUseRPC set")
+		}
+	} else {
+		if c.ConsensusExecutionUseRPC {
+			if !c.Execution.RPCServer.Enable {
+				return errors.New("ConsensusExecutionUseRPC is set but execution node has not enabled rpc server")
+			}
+			if c.Execution.ConsensusRPCClient.URL == "" {
+				return errors.New("ConsensusExecutionUseRPC is set but execution node has empty consensusRPCClient url")
+			}
+			if !c.Node.RPCServer.Enable {
+				return errors.New("ConsensusExecutionUseRPC is set but consensus node has not enabled rpc server")
+			}
+			if c.Node.ExecutionRPCClient.URL == "" {
+				return errors.New("ConsensusExecutionUseRPC is set but consensus node has empty executionRPCClient url")
+			}
+		} else {
+			if c.Execution.ConsensusRPCClient.URL != "" {
+				return fmt.Errorf("consensus and execution are configured to communicate directly but execution node has non-empty consensusRPCClient url: %s", c.Execution.ConsensusRPCClient.URL)
+			}
+			if c.Node.ExecutionRPCClient.URL != "" {
+				return fmt.Errorf("consensus and execution are configured to communicate directly but consensus node has non-empty executionRPCClient url: %s", c.Node.ExecutionRPCClient.URL)
+			}
```

Comment:
nitpick: those checks are fine 🙂, but not necessary, we can just ignore those RPC configs if Consensus and Execution are in the same process and ConsensusExecutionUseRPC is not set.